src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/EditWalletDto.java

```diff
@@ -0,0 +1,11 @@
+package pl.byczazagroda.trackexpensesappbackend.dto;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+
+public record EditWalletDto(@NotNull @NotEmpty @Min(value = 1) long id, @NotBlank @NotEmpty @Size(max = 20) @Pattern(regexp = "[a-z A-Z]+") String name) {
```

Comment:
Just like @pyanik said primitive variable had to have value, so no point to type validations (NotNull, NotEmpty) on "id" parameter.

Reply:
**Update:**
**Please do not remove validation annotations from this argument.** Change type of it from `primitive long` to `objective Long`. So in this case we will be sure it's validation controlling flow and we are catching this type of exception in our proper place.

Reply:
Done - changed to objective Long.